test/attr/ignore_deprecation_warnings.swift

```diff
@@ -0,0 +1,95 @@
+// RUN: %target-typecheck-verify-swift
+
+@available(*, deprecated)
+func deprecatedFunc() {}
+
+@available(*, deprecated)
+func deprecatedFuncWithReturnValue() -> Int {
+    return 3
+}
+
+@available(*, deprecated)
+func deprecatedFuncWithReturnValue2() -> String {
+    return ""
+}
+
+@available(*, deprecated)
+var deprecatedVariable = ""
+
+@available(*, deprecated)
+struct deprecatedType {}
+
+
+@ignoreDeprecationWarnings 
+// FIXME: deprecation warnings are still diagnosed for top-level variable declaration
+// this has to do with Swift diagnosing deprecation warnings for deprecated top-level variables which are assigned to a deprecated property
+var topLevelDecl = deprecatedVariable // expected-warning {{'deprecatedVariable' is deprecated}}
+
+@ignoreDeprecationWarnings
+func x() {
+    deprecatedFunc()
+}
```

Comment:
Unrelated to the change you just did, but should we also have a test to make sure a deprecated type in parameter position gets silenced by the attribute on the function itself? e.g:

```swift
@ignoreDeprecationWarnings
func foo(_ x: DeprecatedType) {}
```

Also FWIW type names should generally start with uppercase letters (in both this test and the other one). It's not a big deal tho.

Reply:
There